frontend/src/pages/ImportCkan/SelectGroupsScreen.jsx

```diff
@@ -0,0 +1,589 @@
+    updateGroup(currentIndex, grp);
+  };
+
+  const updateName = (n) => {
+    const grp = currentGroup;
+    grp.openDataMetadata.metadataTypeName = n;
+    updateGroup(currentIndex, grp);
+  };
+
+  const updateDescription = (d) => {
+    const grp = currentGroup;
+    grp.openDataMetadata.metadataTypeDescription = d;
+    updateGroup(currentIndex, grp);
+  };
+
+  const onCategoryChange = (c) => {
+    const categoryUuid = c.currentTarget.value;
+
+    const grp = currentGroup;
+    grp.openDataMetadata.metadataTypeCategory = categoryUuid;
+    updateGroup(currentIndex, grp);
+  };
+
+  const updateResourceName = (index, value) => {
+    const grp = currentGroup;
+    grp.resources[index].name = value;
+    updateGroup(currentIndex, grp);
+  };
+
+  const handleResourceMetadataChange = (index, name, value) => {
+    const grp = currentGroup;
+    grp.resources[index].openDataMetadata[name] = value;
+    updateGroup(currentIndex, grp);
+  };
+
+  let conversionStrategyGui = (
+    <div>
+      <h2>Create new metadata type</h2>
+      <Input type="text" value={currentGroup.openDataMetadata.metadataTypeName} onChange={(e) => updateName(e.target.value)} placeholder="Name" required />
+      <TextArea value={currentGroup.openDataMetadata.metadataTypeDescription} onChange={(e) => updateDescription(e.target.value)} placeholder="Description" required />
+      <RootLevelCategorySelector onChange={onCategoryChange} />
+    </div>
+  );
+
+  if (currentGroup.openDataMetadata.strategy === 1) {
+    conversionStrategyGui = (
+      <div>
+        <h2>Select metadata type</h2>
+        <Select name="metadataTypeName" value={currentGroup.openDataMetadata.metadataType} onChange={handleMetadataTypeChange} required>
+          <option value="" disabled>Metadata type</option>
+          {
+            Object.keys(metadataTypes).map((key) => (
+              <option key={key} value={key}>{metadataTypes[key].name}</option>
+            ))
+          }
+        </Select>
+      </div>
+    );
+  } else if (currentGroup.openDataMetadata.strategy === 1) {
+    conversionStrategyGui = (
+      <div>
+        <h2>Select metadata</h2>
+        <p><i>All resources will be appended to the selected metadata.</i></p>
+        <Select name="metadataName" value={currentGroup.openDataMetadata.metadata} onChange={handleMetadataChange} required>
+          <option value="" disabled>Metadata</option>
+          {
+            myMetadatas.map((metadata) => (
+              <option key={metadata.uuid} value={metadata.uuid}>{metadata.name}</option>
+            ))
+          }
+        </Select>
+        <p>
+          You have
+          {' '}
+          {myMetadatas.length}
+          {' '}
+          metadatas
+        </p>
+      </div>
+    );
+  }
+
+  return (
+
+    loadingState !== -1 ? (
+      <div>
+        <p>Please wait, loading database state</p>
+        <ProgressBar style={{ width: `${(loading / total) * 100}%` }} />
```

Comment:
If we want to avoid inline styles, this can be achieved by styled-component props (see https://styled-components.com/docs/basics#adapting-based-on-props)

Reply:
Will fix tomorrow

Reply:
i ended up using attrs as props generated new styles. 200 styles later, styled-components begun complaining